docs/articles/documentation/reference/test-api/testcontroller/dispatchEvent.md

```diff
@@ -0,0 +1,214 @@
+
+### Custom Events
+
+Custom Events are all events that are neither [Mouse](#mouse-events), [Keyboard](#keyboard-events), [Input](#input-events), [Focus](#focus-events) nor [Pointer](#pointer-events) events.
+
+TestCafe calls the [CustomEvent Constructor](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) to create a custom event. Custom events support [CustomFocusEvent Properties](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent#properties) in `options`.
+
+In the example below, `foo` is passed as `eventName` to `options`. Since TestCafe can't recognize the event, it uses the CustomEvent constructor.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture`Custom Events`
+    .page('www.example.com');
+
+test('Dispatch a CustomEvent', async t => {
+    const target  = Selector('#target');
+
+    await t
+        .dispatchEvent(target, 'foo')
+});
+```
+
+You can tell TestCafe which constructor to use. See [Explicit Constructor Assignment](#explicit-constructor-assignment).
+
+## Options
+
+Use the `options` object to specify parameters for events fired with `t.dispatchEvent`.
+
+All DOM events support the following `options` properties:
+
+* [Event Interface Properties](https://developer.mozilla.org/en-US/docs/Web/API/Event#properties)
+
+Unless otherwise specified, events have `bubbles` and `cancelable` options set to `true`.
+
+Depending on the `eventName` parameter, a different constructor is called for an event and different sets of additional `options` are available. See [Event Types](#event-types).
+
+You can explicitly set a constructor for an event. See [Explicit Constructor Assignment](#explicit-constructor-assignment).
+
+### Explicit Constructor Assignment
+
+You can explicitly tell TestCafe which event constructor to use. Pass the `eventConstructor` property to `options` (for example, `eventConstructor: 'TouchEvent'`).
+
+The following example fires a [non-cancelable](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable), [non-bubbling](https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles) `TouchEvent` on `target`.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture`Dispatch Events`
+    .page('./index.html');
+
+test('Dispatch a TouchEvent', async t => {
+    const target    = Selector('#target');
+
+    const eventArgs = {
+        cancelable: false,
+        bubbles:    false
+    };
+
+    const options = Object.assign(
+        { eventConstructor: 'TouchEvent' },
+        eventArgs
+    );
+
+    await t
+        .dispatchEvent(target, 'touchstart', options)
+});
+```
+
+Because `eventConstructor: 'TouchEvent'` is passed to `options`, TestCafe calls the `TouchEvent` constructor to create the event. You can use [TouchEvent Properties](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent#properties) in this event's `options`.
```

Comment:
Did you mean "from this event's options"?

Reply:
No, more like:

> You can use TouchEvent Properties as `options` for this event.

Reply:
yes that works

Reply:
```suggestion
Because `eventConstructor: 'TouchEvent'` is passed to `options`, TestCafe calls the `TouchEvent` constructor to create the event. You can use [TouchEvent Properties](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent#properties) as `options` for this event.
```